### PR TITLE
MF-313 - Implement basic Cassandra reader

### LINKED ISSUE #313: Implement basic Cassandra reader
A basic reader that only reads messages and has a pagination should be implemented to facilitate experimenting with and demoing Mainflux system.

Reader exposes only one HTTP REST API endpoint to read messages from DB.

Something similar existed before for MongoDB: https://github.com/MainfluxLabs/mainflux-mongodb-reader

### PR DESCRIPTION
**Pull request by anovakovic01**
This pull request resolves #313.

**Summary:**
- Add common reader interface and shared HTTP API
- Add Cassandra reader implementation
- Add `docker-compose.yml` for Cassandra reader
- Add tests for HTTP API
- Add tests for Cassandra reader
- Add docker network for core services

### CODE REVIEW

File: readers/api/transport.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	auth = tc
+
+	opts := []kithttp.ServerOption{
+		kithttp.ServerErrorEncoder(encodeError),
+	}
+
+	mux := bone.New()
+	mux.Get("/channels/:chanID/messages", kithttp.NewServer(
+		listMessagesEndpoint(svc),
+		decodeList,
```
Comment (dborovcanin) [inline readers/api/transport.go:43]: Rename to `decodeRequest` or `decodeReq`.

File: readers/api/transport.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		}
+		return err
+	}
+
+	return nil
+}
+
+func getQuery(req *http.Request, name string, fallback uint64) uint64 {
+	vals := bone.GetQuery(req, name)
+	if len(vals) == 0 {
```
Comment (dborovcanin) [inline readers/api/transport.go]: Multiple limit and offset query params (`...&limit=3&offset=4&limit=32`) should cause an error.

File: readers/messages.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	"errors"
+
+	"github.com/mainflux/mainflux"
+)
+
+// ErrNotFound indicates that requested entity doesn't exist.
+var ErrNotFound = errors.New("entity not found")
+
+// MessageRepository specifies message reader API.
+type MessageRepository interface {
```
Comment (dborovcanin) [inline readers/messages.go:13]: We can rename `MessageReposistory` (reader as well as writer) to `Reader` and `Writer`.

**Review by dborovcanin — CHANGES_REQUESTED** [review]
README.md and API documentation is missing, but those could be a part of another noissue pull request.

**Comment by anovakovic01** [comment]
@dusanb94 I put WIP label for a reason. This PR is not ready for review. Don't waste your time with this until it is ready.

**Comment by nmarcetic** [comment]
LGTM! 
I have two small remarks. You changed the docker network to `docker_mainflux-base-net` but didn't changed the network in `docker/addons` so I suppose that addons like influx, mongo etc... will not work with core composition. The second one is README file, I don't see any README also I think `offset` and `limit` should be mentioned in README also their default values (if not sent as query param what will happen). 

Nice job @anovakovic01  ;)

**Review by nmarcetic — APPROVED** [review]

File: readers/api/transport.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+		limit:  limit,
+	}
+
+	return req, nil
+}
+
+func encodeResponse(_ context.Context, w http.ResponseWriter, response interface{}) error {
+	w.Header().Set("Content-Type", contentType)
+
+	if ar, ok := response.(mainflux.Response); ok {
```
Comment (dborovcanin) [inline readers/api/transport.go:86]: Rename `ar`.

**Review by dborovcanin — CHANGES_REQUESTED** [review]

File: readers/api/transport.go (hunk 4 above)
Comment (anovakovic01) [reply readers/api/transport.go:86]: This variable is called like this everywhere, so I don't see a reason for that. If you want to change it, send separate PR where you will update it in other services too.

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**